scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -51,6 +51,20 @@ trait Config extends Serializable {
     }
 
   /**
+   * Add files to be localized to the config. Intended to be used by user code.
+   * @param cachedFiles CachedFiles to be added
+   * @return new Config with cached files
+   */
+  def addDistributedCacheFiles(cachedFiles: CachedFile*): Config =
```

Comment:
I mostly agree with you, except for using the same code for Job & Execution API. They seem so different at this point that I don't think that matters.

Anyway, while it's conceptually something in the Config, I think the fact that the current API requires access to the `Mode` couples it to the run environment. So I'm not worried about that either. I agree that the API is not great for multiple files, I think this is acceptable:

```scala
Execution.zip(file1Ex, file2Ex, file3Ex).map { case (file1, file2, file3) =>
  // ...
}
```